### DISCUSSION TITLE
[3.2.0.dev1] User Experience and Accessibility Improvements

### DISCUSSION THREAD

**Discussion by slook** [nicotine-plus/nicotine-plus#1526] [Ideas]
Concepts for minor changes to the existing components of the User Interface 

- This is not a conversation thread, and as such, please keep replies inline and on-topic.
- Nothing here requires immediate reply nor action.
- These suggestions are in no particular order.

**Comment by slook** [comment]
DONE in Nicotine+ 3.2.0 - **The option 'Show Flag Column in User Lists' does not need to be in the View menu.**

Perhaps it can be moved into the Preferences, or removed entirely, as there is now context-menus which provide adequate function in all TreeViews. I can't see any benefit having flags off.

**Reply by mathiascode** [reply]
test reply

**Reply by slook** [reply]
dont forget also to remove 'Show Flag Column in User Lists' from the Keyboard Shortcuts dialog.

**Reply by x1337ctrl** [reply]
just stupid question is nicotine has option to resume downloads or pause it like Utorrent?

**Reply by slook** [reply]
Abort and Retry serve this purpose @x1337ctrl - Nicotine+ will try to resume a partially completed download if it can.

This is off-topic. I have raised https://github.com/nicotine-plus/nicotine-plus/discussions/1526#discussioncomment-1311163 to address this valid point about the confusing terminology used.

**Reply by mathiascode** [reply]
https://github.com/nicotine-plus/nicotine-plus/commit/03dac14c221514abd4e67bc057bfdbd2275585a7

**Comment by slook** [comment]
Edit: FIXED in Nicotine+ 3.2.0

_The following string should be changed from:_
"You can create a new account by entering your desired details below. Please keep in mind that some usernames may already be taken. If you're unable to connect with your selected username, try choosing another one. If connecting remains a problem, verify the settings below and those of your internet connection."

_To:_
"You can create a new account by entering your desired details below. Please keep in mind that some usernames may already be taken. If you're unable to connect with your selected username, try choosing another one. If connecting remains a problem, verify the settings below."

(remove "and those of your internet connection.")

Some new users believe that they need to put the login details for their ISP or something in the text boxes directly below this string.

We are not responsible for providing support for user's internet connections, and it is patronizing to offer it.

**Reply by mathiascode** [reply]
I forgot this string existed... It should also be updated to mention **Soulseek** account, like the setup assistant.

(These sub-discussions seem fine, I don't think we need a bunch of new threads for every minor suggestion)

**Reply by slook** [reply]
Well, the text box allows the entry of any URL, so technically it might not always be the case that the client will be used to connect to a Soulseek account. It's possible there might be a good reason why this word has been omitted from the string.

**Reply by mathiascode** [reply]
Since the network is referred to as Soulseek (and there are no other Soulseek servers around) I think it would reduce ambiguity, if some new users indeed believe it's a login form for some ISP account.

I don't know why the server is the first text entry on the page, I would move it down to the bottom of the page under an "Advanced" section.

> It's possible there might be a good reason why this word has been omitted from the string.

No good reason, I wrote the string originally.

**Reply by slook** [reply]
Cool, thats okay then. In any case your right it is the slsk:// protocol is always going to be known as the Soulseek network regardless of the name of the official server, so it is then an appropriate modification to the string.

**Comment by slook** [comment]
EDIT: Fixed in 3.2.0

The General catagory in Preferences is currently blank, it could contain the big N+ logo with a "Setup Assistant" shortcut button. _(mockup):_
![nicotine-plus Setup Assistant](https://user-images.githubusercontent.com/88614182/132077420-25c16194-6dd7-427f-933e-15bf4edbe64e.png)
_(mockup image)_

If this idea were going to be implemented, the "General" catagory should be renamed to "Setup", and the button named "Setup Assistant".

A button directly below "Manual Configuration", would jump to the 'Network' category.

**Reply by mathiascode** [reply]
I agree that something could be done about the empty categories, but I don't think this is the answer.

**Reply by slook** [reply]
Something similar to the empty main window tabs would be great (see (https://github.com/nicotine-plus/nicotine-plus/commit/928790445ce1eaa0248c69a15193e318a00c23b0), slightly detailed to provide some useful hints and tips.

- General = [Network Icon] Port forwarding guide, what is UPnP, troubleshooting steps (like page 3 of Setup Assistant)

- Transfers = [^v Icon(s)] Sharing etiquette (like final page of Setup Assistant), disk/partition allocation tips (see https://github.com/nicotine-plus/nicotine-plus/discussions/1530#discussioncomment-1279335)

- Chat = [Chat Icon] Command help (like ? popover), Privacy considerations (logging is local, slsk is p2p but unencrypted)

**Comment by slook** [comment]


**Reply by mathiascode** [reply]
Icon only would make it even more confusing (people don't always read the tooltips).

**Reply by slook** [reply]
True, keep it as it is then. But on the left there is not the good place for it imo. I don't know how much people need this feature, but with the speed of slsk nowadays, I would think it is rare. (Edit: I guess it handy when user offline?)

I was also thinking "_Load" sounds like something heavy. Perhaps "_Open" would indicate the true nature of the function.

The tooltip should explain more detail about why a user might want to save/load/open another user's index from disk.
(see https://github.com/nicotine-plus/nicotine-plus/discussions/1530#discussioncomment-1279455)

The dialog box title "Select a File" could be much longer, if possible.

**Reply by mathiascode** [reply]
> True, keep it as it is then. But on the left there is not the good place for it imo.

We can probably move it.

> I don't know how much people need this feature, but with the speed of slsk nowadays, I would think it is rare.

Someone berated me once because they thought the feature had been removed, so people do use it.

**Reply by slook** [reply]
Got a filing cabinet icon?

To get rid of these confusing arrows, as they are already better used on expand/collapse.

**Reply by mathiascode** [reply]
> Got a filing cabinet icon?

There is one, just need to make sure that the icon theme MX Linux uses doesn't drastically change the appearance. It already changed this:

![](https://user-images.githubusercontent.com/8754153/132132710-73b2c7be-3b68-4d0e-a246-293fe71e6825.png)

to that "open" arrow.

**Reply by slook** [reply]
_Nice one, button looks okay here on MX Linux 18.3 (Papirus-Dark icons)_
![nicotine-plus User Browse Open Share button icon](https://user-images.githubusercontent.com/88614182/132134497-c6adddf9-ab3c-48c3-9d65-1acbf59cfdbf.png)
(I tried some other icon sets and always is the folder icon, perfect)

However, it is still not that obvious how the function is supposed to work. I suggest:

- Rename button caption label to "Open List"
- Rename dialog box title to "Select a previously saved Shares List File" would then be very helpful.

Edit: **FIXED in Nicotine+ 3.2.0**

**Comment by slook** [comment]
Edit: FIXED in Nicotine+ 3.2.0 - **The 'Save Shares List to Disk' has an incorrect Download icon on it:**
![nicotine-plus Save shares list to disk](https://user-images.githubusercontent.com/88614182/132079422-04ae662d-fa53-476e-9a56-d21175e8b3f4.png)

- It should be a floppy disk.

Ctrl+S would be a good key binding for it tho.

It's location is non-intuative, perhaps it should be alongside the 'Load from Disk' button, there are too many downward pointing arrows at this location! Yet none of them actually download anything argh!!!
![nicotine-plus User Browse down arrows](https://user-images.githubusercontent.com/88614182/132079641-65e90236-08ce-4594-b9d6-4bcdaa6ebddd.png)

**Reply by slook** [reply]
TODO: find Floppy Disk icon

**Reply by mathiascode** [reply]
There is none, most icon themes replace it with the download icon.

**Reply by mathiascode** [reply]
I see some icon themes completely screw up the current icon too. Great...

**Reply by mathiascode** [reply]
![](https://user-images.githubusercontent.com/8754153/132110298-eb983ac4-da1e-4c92-b673-ff7c5387bd4b.png)

There is a separate floppy icon, but it looks awful in Adwaita.

**Reply by slook** [reply]
I reckon its okay... floppy disks were never designed to look aesthetically appealing!

I think the button does not belong there, perhaps it would look less awful next to Open From Disk in the header bar?  

| **`Open List`** | **`Save List`** | `Settings` | `Menu` |

Alternatively, move the button to the left of the Find text entry box (such that it is next to Shared: xGiB)

**Comment by slook** [comment]
_FIXED in Nicotine 3.2.0 -_ **Search History is not long enough**

Add an option in Preferences > Searches to allow configuration of the number of previous searches to remember.

Or more than 15, suggest at least 100 would make the AutoComplete much more useful.

It may be necessary to constrain the dropdown listbox height to a reasonable height and add a scrollbar.

_Edit: the auto-completion drop-down confusingly interferes with operation of the Search History feature, see https://github.com/nicotine-plus/nicotine-plus/discussions/1535#discussioncomment-1455269_

**Reply by mathiascode** [reply]
> It may be necessary to constrain the dropdown listbox height

As far as I know, this isn't possible.

**Reply by mathiascode** [reply]
I've been thinking about using something else than GtkComboBox for the search/user text entries in the long run. In GTK 3, scrolling is rather painful with a large number of items, and doesn't work at all in GTK 4.

**Reply by mathiascode** [reply]
In any case, the limit is now 200.

**Reply by slook** [reply]
> scrolling is rather painful with a large number of items

I see what you mean, but it is workable.

**Comment by slook** [comment]
**Text-to-Speech requires a Test button in Preferences or converting to a plugin**

Accessibility issue. TTS is challenging to setup if a custom command is required. Users of this feature are likely to have very specific needs in regards to their speech synthesizer software. Therefore, a Test Speech playback button is essential.

_Alternative Idea: Further development by converting the TTS feature to a plugin:_

After setup, during actual usage, playback can often get behind if the messaging activity is too busy. Toggling the built-in TTS toolbar button does nothing after messages have been queued in the speech synthesizer, so there is no way to stop it/skip it (other than restarting the application). This problem can make the session unusable if there is a spam attack or suchlike. Most TTS synthesizers have a stop/reset feature, so the TTS chould be improved to support that.

As the output format is punctuated, we could implement [extra logic during playback to improve the coherence of ongoing conversations](https://github.com/slook/nicotine-plus/commit/5d15f25a126617d0406bc5e7801d667a1db74f90):

- It is not necessary to keep repeating the part before the comma ("In room %(room)s,") if the subsequent following message originates from the same room, and also;
- the username part between the comma and the colon (", user %(user)s said:") can also be skipped if the same user is posting a multi-line message, unless there is a pause (of say, over 5 seconds or so).
- if there is no punctuation in the speech format string, then just fallback to announcing the whole chat line in its entirety.

There might be plugin applications that could benefit from a [toolbar Button or CheckButton in the chat interface](https://github.com/slook/nicotine-plus/commit/7545854ebbfcbea77e05f415f0be541eb9d89876), at least to trigger or toggle a plugin on/off per tab, or possibly for some other custom function (such as doing some process on the last chat line, in this case, to announce it). Therefore, we could consider converting the TTS feature into a plugin as a test case for plugin-owned chat buttons.

(also the Chat Command Help button could be converted to be a plugin as well, for instance plugins might want to create popovers to configure setting(s) or to display some output data statistics or other informational output, without printing it into the chat view).

**Reply by slook** [reply]
My students use "eSpeak", so I put "espeak $" but it does not work at first (requires restart).

**Reply by slook** [reply]
see also _DONE_ - https://github.com/nicotine-plus/nicotine-plus/discussions/1526#discussioncomment-1294119

**Comment by slook** [comment]
**Chat AutoComplete**

- [ ] Better to start with Usernames, instead of RoomList items
- [ ] Support /join for suggesting entries Room List

**Comment by slook** [comment]
**Interests tab Recommendations ListView needs a left-click function** (in addition to right-click)

- Single-click to show context-menu (both left and right buttons).

- _DONE in Nicotine 3.2.2_ - Double-click on Recommendation could trigger default action "Recommendations for Item".

- Drag and Drop would be ideal into Like/Dislike lists.


Edit: see https://github.com/nicotine-plus/nicotine-plus/issues/1730

**Comment by slook** [comment]
EDIT: Fixed in 3.2.0

User Browse does not remember focus of list, so if a Filename is selected then it is easily forgotten when switching tabs, because when up/down is pressed then the folder is navigated unexpectedly.

**Comment by slook** [comment]
**Sub-Menus Popovers are not easily accessible via keyboard, and Main Menu contains too many items**

Consider move all Shares items into separate "Shares" menu, to match the native menus.

Then, it would be ideal if all sub-menus could be activated directly with `Alt+V`, `Alt+S`, `Alt+M`, `Alt+H` without needing to open main menu first, if possible to activate menu popovers in this way.

Edit: Also `Alt+F` for the Main Menu popover (in addition to `F10`) now that all the `F` ups are resolved, burger menu should ideally match native menu navigation as close as possible.

**Reply by mathiascode** [reply]
> burger menu should ideally match native menu navigation as close as possible

I don't agree. The burger menu should prioritize important actions, and given the number of questions about shares, in addition to people being unaware of the "Modes" menu (will be removed eventually), adding all the shares-related items under a sub-menu isn't something I feel comfortable doing.

The current number of items is reasonable in my opinion. I've attempted to reduce the number when possible, and something like Firefox's menu is roughly the same size.

**Reply by slook** [reply]
Thank you for your explanation - that is fair reasoning - in which case:

- It would be ideal if the main burger menu could be opened with `Alt+F` and `Alt+S` and `F10` ;
- also each sub-menus could be opened directly with `Alt+V`, `Alt+M`, `Alt+H` (without having to press F10 first)...

...because proper keyboard accessibility to the menus is currently only possible with the header bar disabled.

**Comment by slook** [comment]
_BLOCKED_ - **Remove "Clear All Results" menuitem from the search tab context-menu.**

It does not seem to serve any useful function, since a new search will open a new tab anyway, and there is no refresh results function.

**Reply by mathiascode** [reply]
I want to keep it until we store filters for wishlist searches.

**Comment by slook** [comment]
_RESOLVED in Nicotine+ 3.3.0.dev4_ - **Add option to "Deactivate search result filters when Result Filters bar is hidden (Alt+R)"**

as a checkbox in Preferences > Searches > Search Result Filters > 

It would be ideal if the text entry boxes were still populated with the last entries when it is shown again. This would close #1400 ...
_"I miss being able to turn off the Result Filters, not just hide them, as used to be possible via the main UI. Maybe the "Results Filters" button could be given a Ctrl + click feature to toggle their on/off state, instead of the default plain click which currently just toggles their visibility."_

**Comment by slook** [comment]
FIXED in Nicotine+ 3.2.0

**Edit Virtual Name entry should be pre-filled with the current Virtual Folder Name**

Alternatively, in-cell editing instead of dialog box.

Preferences > Shares > Rename

**Reply by mathiascode** [reply]
https://github.com/nicotine-plus/nicotine-plus/commit/d832c0a9ff0b62e9166576e44bc6c8d8113ffd9f

**Comment by slook** [comment]
Edit: Feature already present in GTK 3.24+ (Ubuntu 18.04, Debian 10)

**Smiles for the Chat**  

someone somehow typed: 😀️ a palette popover of special characters could be cool.

![nicotine-plus Chat Smilies](https://user-images.githubusercontent.com/88614182/132268334-5de35940-3bd3-42c2-a752-77ec6fb5e06c.png)

**Reply by mathiascode** [reply]
Right-click text entry -> "Insert Emoji" is provided by GTK, if your GTK version is recent enough.

**Reply by slook** [reply]
Wow, the technology... Then maybe I should sell my fax machine 🖨️  😂️

**Comment by slook** [comment]
_DONE in Nicotine 3.3.0_ - **Private Chat: Toggle Text-to-Speech**

Cannot be disabled per chat tab, which I would imagine would be highly annoying if using text-to-speech. (Edit: I've tried it, yes it is highly annoying!).

The omission of a toggle button makes the Text-to-Speech feature totally useless unless very few chats are open.

**Reply by slook** [reply]
This is tested with eSpeak, which is about the worst text-to-speech engine available.

DONE - **The default Private Chat string `%(user)s told you.. %(message)s` announces the "dot dot"..**

DONE - It needs to be changed to `%(user)s told you, %(message)s` (with a comma, instead of two dots).

DONE - Private Chat Text-To-Speech toggle button

**Comment by slook** [comment]
**Bitrate and other columns are missing from the Downloads tab (and Uploads tab)**

Transfers listed in TreeView do not have 'Bitrate', 'Length', 'Country' columns. These would be particularly useful in the Downloads tab (and also wouldn't do any harm in the Uploads tab.

It would be interesting to see what part of the world the files came from / are going to, if 'Country' data were available at the point of Transfer.

Additionally, 'Artist' and 'Title' columns would be useful for an option to display ID3 tags for local files that have been Downloaded.

Perhaps, the existing tinytag implementation can be extended to provide additional functionality, without slowing performance.

**Comment by slook** [comment]
DONE in Nicotine+ 3.2.0 - **File Properties menuitem lacks standard Alt+Enter shortcut key binding**  ( https://github.com/nicotine-plus/nicotine-plus/commit/8dcb4cef1b165b00237eb7410e5fea3072a8812b + https://github.com/nicotine-plus/nicotine-plus/commit/4331ba99dc53d912d6a8f0d9227e2ebb174c3d69 )


Also, this menuitem belongs at the bottom of the context-menu.

**Comment by slook** [comment]
DONE in Nicotine+ 3.2.0

**GTK emoji picker (Ctrl + ".") in Keyboard Shortcuts**

In the "Editing" section, but only if GTK version >= 3.24

See #1556 and https://github.com/nicotine-plus/nicotine-plus/commit/d92bf58326c2a87eb75ea407f771391bbe680400

**Comment by slook** [comment]
_WontFix: This is a problem with the Papirus icon package._

**Chat Rooms icon should depict a speech bubble**

The exact same icon is used for User Info, so one of the icons needs to be changed.

And/or, an information ( _i_ )  icon could be used for the User Info tab instead.

**Reply by mathiascode** [reply]
Depends on the icon theme.
![](https://user-images.githubusercontent.com/8754153/132954536-9d4af0c0-660f-4b74-8730-71874fd10c4d.png)

**Reply by slook** [reply]
Icon Theme = Papirus-Dark (and all other Papirus icon sets)

![nicotine-plus Old GTK theme Chat icon](https://user-images.githubusercontent.com/88614182/132954631-3e7f7e3f-c1dc-4d11-9d77-387b3d89a634.png)

![nicotine-plus Old GTK theme Chat icon large](https://user-images.githubusercontent.com/88614182/132954654-3d019c5a-9fd2-4cd3-ab8c-9f4edc925fc7.png)

**Comment by slook** [comment]
DONE in Nicotine+ 3.2.0

**Retry button+menuitem should be labelled `Resume` to match the new description for the Downloads tab**

This function does indeed 'Resume' from an interrupted download from the point where it left off.

Retrying a Finished download does not seem to be possible, so the button and menuitem should ideally be disabled/greyed-out or invisible for those cases.

Additionally, since 'Aborted' downloads can infact be resumed, then perhaps **`Stop`** and "Stopped" would be less harsh terminology that better indicates the flexibility of the client, because users may well think that pressing 'Abort' will delete item(s) from the list forever.

**Reply by mathiascode** [reply]
`Abort` is a rather technical term from the early days. Deluge and qBitTorrent seems to use `Resume` and `Pause`.

**Reply by slook** [reply]
However @mathiascode `Pause` does not include the letter `T` so changing the string to that word would involve key binding reassignment.

I therefore suggested `sTop` would be easier to implement for you and the translators. Also `R` and `T` are adjacent on a QWERTY keyboard so this is already user friendly and easy to remember the way it is.

_Edit:_ Additionally, if unifying button labels with the Uploads tab is desired in future, then that Abort function does more than just Pause the remote user, rather it does indeed put a complete `Stop` to the upload (and the upload cannot be Resumed locally).

**Reply by mathiascode** [reply]
I don't think `Stop` will reduce the number of questions about where the pause functionality is. I prefer that we use `Pause` to follow conventions of other file sharing programs. I could add another `P` key binding that does the same as `T`, if necessary, since the uploads tab will still keep the `Abort` string (at least for now).

**Reply by mathiascode** [reply]
I'm aware that Tixati and potentially some other programs use `Stop`, but SoulseekQt (where most users would come from) uses `Pause`.

**Reply by slook** [reply]
I wasn't aware that SoulseekQT uses `Pause` in which case your suggestion is most appropriate then.

Also "Paused" status instead of "Aborted", and "Paused" items in the Clear Downloads menu.

**Reply by slook** [reply]
![nicotine-plus Download buttons new](https://user-images.githubusercontent.com/88614182/132966276-248dc444-0fd5-494b-8a90-9509c5f782b7.png)

https://github.com/nicotine-plus/nicotine-plus/commit/feb630abafd77110fc956d0068ac49ce0fac923b looks good however context-menu mnemonic R for "_Resume" is missing, and "Paus_e" since P is used for Player.

**Reply by mathiascode** [reply]
Any suggestion for `File Properties`? There was a conflict previously (`File P_roperties` and `_Retry`).

**Reply by mathiascode** [reply]
`R_esume`
`P_ause`

`R_etry`
`_Abort`

perhaps?

**Reply by slook** [reply]
`_Resume`
`Paus_e`

`_Retry`
`Abor_t`

`I` for `F_ile Properties` seems appropriate because it is `i`nformation about the file?

Whatever happens, R definitely has to be for `_Resume` and  `_Retry`

**Reply by slook** [reply]
https://github.com/nicotine-plus/nicotine-plus/commit/e93cea20770406733df5a7b97da02c47da593375 is plenty good enough, `a` is fine for the context-menu mnemonic for P_ause as it doesn't conflict with anything now. 

Good call on the `Pause` button label and icon, nobody can miss that now!

**Comment by slook** [comment]
**Options for setting background color of all TreeView Lists and Chat TextView**

- Preferences > User Interface > Lists > "List background color"

- Preferences > User Interface > Chats > "Room background color"

- Preferences > User Interface > Chats > "Highlighted text background color"

At the very least, BG option is required for readability on some dark themes (accessibility issue).

In a perfect world, it would be ideal if all Chat text items had both ForeGround and BackGround colors that were fully configurable, such as to set links and username tags in colored blocks for example.

_Additionally, please move the Text Entries section above the Chats section of the Preferences page._

**Reply by mathiascode** [reply]
> At the very least, BG option is required for readability on some dark themes (accessibility issue).

Out of curiosity, why use these broken themes?

**Reply by slook** [reply]
On many OS there isn't always an easy way to change or customize the GTK theme (Windows for example). Understandably, some users [don't know how](https://github.com/nicotine-plus/nicotine-plus/issues/1517#issuecomment-917703099) or don't bother changing the default theme, and look at MX 19.4 for example, it's awful.

For those with a choice, it seems people do not always select wisely. Perhaps everything looks great in some other application that is more of a priority on the same installation. Those with a specific learning difficulty (such as dyslexia for example) seem to consider only the font rendering and disregard the UI elements. At my work, users often end up choosing very obnoxious looking desktop appearance preferences.

**Reply by mathiascode** [reply]
It's unfortunate that environments ship with bad defaults. At times it feels like random colors are thrown together without any thought, or ugly hacks are applied that end up hurting the user experience. I know this work is done by people in their spare time, possibly without much experience in design, but I'm becoming increasingly tired of themes attempting to be clever.

I'm not against re-adding these color options to the settings, the theme-related code was rewritten to allow for easier additions after all. It needs to be done without breaking the UI, like the previous implementation did (it applied the background to text/row selections as well).

**Reply by slook** [reply]
I totally agree. It's a world where there are no set-in-stone rules and devs seem to have carte blanche to throw paint at wall, willy nilly. It is not just the hobbists responsible for the ugly hacks. I mean take the option in KDE System Settings > Colors > Options > "Apply colors to non-Qt applications" (on by default) which affects whatever the chosen GTK theme sets, surely practices like these create a lottery. I can understand why mainstream GTK apps such as Opera force their own theme, whilst I am not a fan of this practice either.

The request came from a user in nicotine chat, I think the main criticism is that the Chat TextView doesn't look very 'fun' by modern standards. Many new users don't remember what IRC is, nor do they appreciate it's paradigm. Admittedly, the color of the room background being the same as the N+ main app control surface color does make it look somewhat bland. My preference is for the Chat and List backgrounds to be either Black or White (depending upon if Dark Theme is preferred), but perhaps some of the wannabe script kiddies would love to set some gaudy custom colors for posting screenshots to their Instagram or whatever the fashion is these days.

This leads me to consider if separate default Text colors are required for light vs dark theme. It is a more complex issue than I had imagined.

**Comment by mathiascode** [comment]
_Done in Nicotine+ 3.2.0_

Is the "Show Buttons in Transfer Tabs" option necessary?

It has existed for many years, but I'm not sure if it provides any benefit.

**Reply by slook** [reply]
I'm glad you've mentioned that, you're correct, there is no benefit to having the Buttons hidden, so the option may as well be removed.

It is annoying when the bar disappears because it's shortcut key can be activated anytime.

**Comment by redactedscribe** [comment]
Not sure if this right place, but something that has always bugged me about the UI is the Uploads/Downloads view (which I believe if effectively the same as the grouping mode "Group by Folder" for the Search Files tab).

It's unnecessarily noisy to me and just a bit confusing to look at. For example:
```
...
------- v UsernameA         <filename>      100%
v UsernameB                                 25%    200 KiB/s
--- v UsernameB     <path>  <#> files       25%    200 KiB/s
------- v UsernameB         <filename>      99%    200 KiB/s
------- v UsernameB         <filename>      0%
------- v UsernameB         <filename>      0%
------- v UsernameB         <filename>      0%
v UsernameC                                 100%
...
```

Is there anything that can be done to visually delineate sections and/or simplify the look?

The first row "UsernameB" is mostly redundant if only downloading _1_ folder from a user: we see the same status, percent, size, transfer speeds, time elapsed etc for both the first and second row for UsernameB. Maybe it would be too fiddly to manage technically, but combining the first two rows if only 1 folder is downloaded would help from the UX perspective I believe. Otherwise, it remains visually noisy, and at times confusing; it's been more than once that I've thought something has been wrong when it seemed as if a user was downloading faster than the speed limit I have set, i.e. seeing 200 KiB/s three times could be wrongly assumed as a user downloading at 600 KiB/s. This mistake is easy to make as the transfer speed shown for the root row isn't always identical to the transfers speed on the row of the file being downloaded, which is quite odd to begin with (assuming only 1 folder being download).

But maybe all that is needed is a bit of colour. A subtle but noticeable darker grey background highlight to the rows that are folders could help with visual clarity, and break up the contiguous-list look.

Also, the colour of the percent's background (blue at least on Windows) for the folder rows could be different to the rows for files (for example, only folder rows would have a blue background, with file rows being less saturated / grey). Benefit here is that it'd be a visual indicator for the type of row (file/folder), without having to look to the left of the row to determine that. It would also help break up the duplicity of information (seeing the same styled "100%" on every row is adding to the contiguous look, when in fact the rows aren't all related).

Similarly, text for file rows that are Aborted could be slightly dimmed (similar to non-immediate "N" download users on the Search Files tab), which would help notice them more easily, instead of all the text currently having the same visual weight, regardless of file status.

I'm not sure how flexible these GTK column/row elements are, but any improvement regarding the displaying of uploads/downloads to make them easier to understand at a glance would be appreciated.

**Reply by mathiascode** [reply]
No more treeview-related changes will be included in 3.2.0, and any future changes to them will require `pynicotine/gtkgui/widgets/treeview.py` to be refactored into a class.

**Comment by hboetes** [comment]
_DONE in Nicotine+ 3.2.0_

### Comments are public statements, notes are private (until you share them)

The current code uses the term 'comment' where the term 'note' is more appropriate, since comments are public statements and notes are private jots.

Of course, simply changing the translatable strings would fix the visible part, but don't I think @mathiascode likes that. =)

```
#: pynicotine/gtkgui/userlist.py:569                                                                                                                                          
msgid "Edit Comments"
msgstr ""

#: pynicotine/gtkgui/userlist.py:570                                                                                                                                          
#, python-format                                                                                                                                                              
msgid "Add a few notes associated with user %s:"
msgstr ""
```

**Comment by 82d** [comment]
highlighting friends' nicknames with color
in the "User Inteface" settings, you can choose the colors of List text color. it would be very useful to choose a color for highlighting nicknames that are in the friends list. this would facilitate many processes, especially with the elimination of leeches.

**Comment by slook** [comment]
_DONE in Nicotine 3.2.0 -_ **Allow the linux version to use network locations as folders**

```
2021-11-04 14:24:38 [r_] Can i make a request lol
2021-11-04 14:24:46 [l_] sure?!
2021-11-04 14:24:59 [r_] Allow the linux version to use network locations as folders like vanilla soulseek on windows
2021-11-04 14:25:28 [r_] I keep all my music on my server and before switching to N+. I was able to just point it to the network location and it would work fine
2021-11-04 14:25:54 [l_] Is the network location not mounted into a folder in the filesystem?
2021-11-04 14:26:02 [r_] https://i.imgur.com/nCwiP6E.png
2021-11-04 14:26:12 [r_] This is how I did it on windows on og ss
2021-11-04 14:26:34 [r_] I have that mounted as a drive now for Nicotine+ on windows but I haven't gotten around to it on Linux yet
2021-11-04 14:26:42 [l_] So how do you ordinarily access your network share in linux?
2021-11-04 14:26:54 [r_] Just using the network tab in Dolphin
2021-11-04 14:27:00 [r_] I'm on KDE Manjaro
2021-11-04 14:27:16 [l_] Oh okay, so it is a Samba share?
2021-11-04 14:27:21 [r_] Yes
2021-11-04 14:27:58 [r_] I realize this would probably be better as a feature request instead of screeching at a dev about it lol
2021-11-04 14:28:07 [l_] Intergation for Samba shares with the // format has always been lacking in linux.
2021-11-04 14:29:07 [l_] No problem! It is interesting to get the feedback as to what the users want.
2021-11-04 14:29:49 [l_] Without the screeching then there is no motivation for development, it is possible that others would want the better networking capability.
2021-11-04 14:30:25 [r_] Feature parity with the original client and all that
2021-11-04 14:30:30 [l_] I agree that this would be desireable, I myself have a NAS and this sort of thing is a real pain.
2021-11-04 14:31:01 [l_] Yes, you are right, that is always long been the aim.
2021-11-04 14:31:57 [l_] and with each version the features are improving, even supassing the Qt client in some areas, but there are things that are still lacking.
2021-11-04 14:32:32 [r_] Thank you for all the hard work!
```

![image](https://user-images.githubusercontent.com/88614182/140335264-21c5d43d-da1e-4ca0-a3c9-77daedb47f52.png)

**Reply by mathiascode** [reply]
FTP and WebDAV work fine on my end, but I don't have any NFS or Samba servers to test right now. Are there any errors or useful output in the Nicotine+ log?

**Reply by slook** [reply]
I don't think this is any sort of error, just a lack of proper LAN support in Gtk. I think the user has a workaround in place via a mapped network drive, but this extra step shouldn't be necessary since the Samba share can be browsed from the OS File Manager.

Sometimes I encounter these sorts of issues when running Flatpaks and apps with sketchy sandboxing arrangements that seems to prevent proper browsing of LAN shares.

From what I gather I think that what is desired is a Network Places item in the Gtk dialog that spawns from Preferences > Shares > Add 

_on MX18 (GTK 3.22) There is no Network Places functionality in the default Gtk dialog that is spawned by N+..._
![nicotine-plus Samba Gtk](https://user-images.githubusercontent.com/88614182/140452231-68c32b8f-a7f9-47bc-b0cf-81a9bb148bf7.png)

_If you compare this to the OS default file browser, Xfce Thunar in this case, this allows built-in Samba navigation:
![nicotine-plus Samba Thunar](https://user-images.githubusercontent.com/88614182/140452416-929435e3-5d8b-4835-b6de-6d140bdc95d7.png)
_

I suppose it is possible to Add a dummy entry and then rename it with a manually pasted network:/// path? Perhaps there is some adjustment to the default Gtk folder selection dialog that can be leveraged, idk, if only there was a Text Entry path bar.

Samba doesn't natively mount to a folder in the Linux filesystem, it is expected to work over TCP/IP

**Reply by slook** [reply]
https://docs.gtk.org/gtk3/method.PlacesSidebar.set_show_connect_to_server.html

"
_Sets whether the sidebar should show an item for connecting to a network server; this is off by default. An application may want to turn this on if it implements a way for the user to connect to network servers directly.

If you enable this, you should connect to the GtkPlacesSidebar::show-connect-to-server signal.

Available since: | 3.10
Deprecated since: | 3.18

It is recommended to group this functionality with the drives and network location under the new ‘Other Location’ item._
"

**Reply by mathiascode** [reply]
I've enabled non-local folders in the file chooser: https://github.com/nicotine-plus/nicotine-plus/commit/fc371d347eff0c620b3998753efe0b26b7ba64ce

I've not been able to test this change yet, so more work is probably required. Any testing is appreciated if you have network shares at hand.

**Reply by slook** [reply]
I don't have any network shares to hand @mathiascode , but it looks like a samba path should be accessible now:

![image](https://user-images.githubusercontent.com/88614182/141024232-234cca58-890d-4ae1-8675-fb241aaf5de0.png)

The fact there is no samba shares to browse is handled gracefully on click of "Windows Network":
![image](https://user-images.githubusercontent.com/88614182/141024040-c956b0f6-a032-467c-b138-eef28e3a9cae.png)

This is expected behaviour, similar to that of the OS file manager. @UnixNight should be able to access a Samba path this way?

**Reply by UnixNight** [reply]
@slook I can test on my Linux box in the next couple of days and let you know
